src/Nethermind/Nethermind.TxPool/Filters/TooExpensiveTxFilter.cs

```diff
@@ -59,10 +50,7 @@ public AcceptTxResult Accept(Transaction tx, TxFilteringState state, TxHandlingO
                 }
             }
 
-            UInt256 affordableGasPrice = tx.CalculateAffordableGasPrice(spec.IsEip1559Enabled, _headInfo.CurrentBaseFee, balance > cumulativeCost ? balance - cumulativeCost : 0);
-
-            overflow |= spec.IsEip1559Enabled && UInt256.AddOverflow(tx.MaxPriorityFeePerGas, tx.MaxFeePerGas, out _);
```

Comment:
this line is no longer needed?

Reply:
I don't understand why we were checking overflow of operation `MaxPriorityFeePerGas` + `MaxFeePerGas`. `MaxPriorityFeePerGas` is a part of `MaxFeePerGas`, as `MaxPriorityFeePerGas <= MaxFeePerGas` is required for tx to be valid. So actually it can prevent us from adding valid transaction if:
MaxPriorityFeePerGas <= MaxFeePerGas
MaxFeePerGas > UInt256.MaxValue / 2
MaxPriorityFeePerGas >= UInt256.MaxValue / 2

It was here probably because of me misunderstanding connections between `MaxPriorityFeePerGas` and `MaxFeePerGas` when implementing first version of this filter.
I will write more tests for that in few minutes.

Reply:
Actually no new tests needed, fees per gas are multiplied by `GasLimit`, which is at least 21000. So no way of rejecting valid tx because of this check. But it's not catching anything too, so it's redundant